metals/src/main/scala/scala/meta/internal/parsing/Trees.scala

```diff
@@ -132,6 +132,11 @@ final class Trees(
         dialect(input).parse[Source]
       }
     }
+  } catch {
+    // if the parsers breaks we should not throw the exception further
+    case _: StackOverflowError =>
+      scribe.debug("Could not parse:\n" + path.readTextOpt.getOrElse(""))
+      None
```

Comment:
Note that in general it is not safe to catch `StackOverflowError`; it may leave the JVM in an invalid state.
- https://github.com/scala/scala/commit/64603653f82082431941cb29ad317b669822f28e
- https://groups.google.com/d/topic/scala-language/eC9dqTTBYHg

Reply:
Hmm... so the only option is to restart the process in that case? :thinking:

Reply:
Ok, so reading the article (thanks @armanbilge !) I think it should safe to catch it here since the parser doesn't have any locks and is single threaded.